### PR TITLE
Renderer debugging: fix r_debugSort; filter r_glDebugProfile messages

### PR DESCRIPTION
**Pull request by slipher**
- make r_debugSort work instead of crashing
- severity-level filtering for GL debug messages

### CODE REVIEW

File: src/engine/renderer/tr_shade.cpp [hunk 1]
```diff
@@ -3143,26 +3143,26 @@ void Tess_End()
 	}
```
Comment (DolceTriade) [inline src/engine/renderer/tr_shade.cpp:3143]: why is it safe to return early here but now below?

File: src/engine/renderer/tr_shade.cpp [hunk 2]
```diff
@@ -3143,26 +3143,26 @@ void Tess_End()
...
 
 	// for debugging of sort order issues, stop rendering after a given sort value
-	if ( r_debugSort->integer && r_debugSort->integer < tess.surfaceShader->sort )
+	bool skip = r_debugSort->integer && tess.surfaceShader != nullptr
```
Comment (DolceTriade) [inline src/engine/renderer/tr_shade.cpp:3146]: this nested indentation is really ugly. I'm almost wondering if it's worth it to split out the cleanup stuff below into its own function :-/

File: src/engine/renderer/tr_shade.cpp (hunk 1 above)
Comment (slipher) [reply src/engine/renderer/tr_shade.cpp:3143]: The most important thing is to zero out the counters: numIndexes etc.

File: src/engine/renderer/tr_shade.cpp (hunk 2 above)
Comment (slipher) [reply src/engine/renderer/tr_shade.cpp:3146]: In an upcoming commit I move the last part to a function like this:
```
void Tess_Clear()
{
	tess.vboVertexSkinning = false;
	tess.vboVertexAnimation = false;

	// clear shader so we can tell we don't have any unclosed surfaces
	tess.multiDrawPrimitives = 0;
	tess.numIndexes = 0;
	tess.numVertexes = 0;

	// This is important after doing CPU-only tessellation with Tess_MapVBOs( true ).
	// A lot of code relies on a behavior of Tess_Begin: automatically map the
	// default VBO *if tess.verts is null*.
	tess.verts = nullptr;
	tess.indexes = nullptr;
}
```

Is that what you mean?

Comment (DolceTriade) [reply src/engine/renderer/tr_shade.cpp:3146]: Yes.

**Review by DolceTriade — APPROVED** [review]

**Pull request merged**